homeassistant/components/litterrobot/binary_sensor.py

```diff
@@ -0,0 +1,91 @@
+"""Support for Litter-Robot binary sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Generic
+
+from pylitterbot import LitterRobot, Robot
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import LitterRobotEntity, _RobotT
+from .hub import LitterRobotHub
+
+
+@dataclass
+class RobotBinarySensorEntityDescription(
+    BinarySensorEntityDescription,
+    Generic[_RobotT],
+):
+    """A class that describes robot binary sensor entities."""
+
+    is_on_fn: Callable[[_RobotT], bool] = lambda _: True
```

Comment:
Add back the mixin dataclass so you don't have to set a default for `is_on_fn` and then this should be good from my vantage point

```python
@dataclass
class RequiredKeysMixin(Generic[_RobotT]):
    """A class that describes robot binary sensor entity required keys."""

    is_on_fn: Callable[[_RobotT], bool]


@dataclass
class RobotBinarySensorEntityDescription(BinarySensorEntityDescription, RequiredKeysMixin[_RobotT]):
    """A class that describes robot binary sensor entities."""
```

Reply:
Well that worked. Mypy can be troublesome.

Reply:
Yeah, I only know cause I had help myself on a previous one, haha